packages/electron/src/electron-main/electron-application.ts

```diff
@@ -376,25 +327,15 @@ export class ElectronApplication {
     protected hookApplicationEvents(): void {
         app.on('will-quit', this.onWillQuit.bind(this));
         app.on('second-instance', this.onSecondInstance.bind(this));
+        app.on('window-all-closed', () => this.requestStop.bind(this));
     }
 
     protected onWillQuit(event: ElectronEvent): void {
-        if (typeof this.stoppingTask === 'undefined') {
-            event.preventDefault();
-            this.stoppingTask = new SyncPromise(this.stop());
-            // Once the stopping process is done, we should quit for real:
-            this.stoppingTask.promise.then(
-                () => app.quit(),
-                () => app.quit(),
-            );
-        } else if (!this.stoppingTask.finished) {
-            event.preventDefault();
-        }
-        // `stoppingTask.finished === true`: let the process quit.
+        this.stopContributions();
     }
 
-    protected async onSecondInstance(event: ElectronEvent, argv: string[], cwd: string): Promise<void> {
-        await this.launch({ argv, cwd, secondInstance: true });
```

Comment:
> Why remove cwd???

Work in progress; otherwise, I could not start the app without a workspace error on macOS.

Reply:
BTW, can you please fix the macOS behavior on your branch, and I'll just rebase this PR from yours and get back the `cwd` support. What you wrote 👆 sounds good to me, a very nice addition, but [it does not work on macOS](https://github.com/eclipse-theia/theia/pull/7964#issuecomment-643187076) 😕 We must verify the bundled app behavior too.

Reply:
> * You can do `cd some/place && app .` and it will open `/some/place` as a workspace because the app knows what's the cwd when you do `app .`
> * Now the app is already running, and you open a new terminal and do `cd somewhere/else && app .`. Here it will trigger the second-instance mechanism and forward the cwd used to invoke the second instance to the already running process. If we don't have the new cwd we cannot resolve `app .` to the correct location.

@marechal-p, VS Code handles it differently, right?
```
cat `which code`
#!/usr/bin/env bash
#
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

function realpath() { python -c "import os,sys; print(os.path.realpath(sys.argv[1]))" "$0"; }
CONTENTS="$(dirname "$(dirname "$(dirname "$(dirname "$(realpath "$0")")")")")"
ELECTRON="$CONTENTS/MacOS/Electron"
CLI="$CONTENTS/Resources/app/out/cli.js"
ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" "$@"
exit $?
```

So the approach you have implemented does not work on macOS; when I start the app from a terminal in dev mode with `yarn`, or I launch the electron backend launch config from VS Code, without specifying any `file` my options will fall back to the electron process: `"/Users/akos.kitta/Desktop/theia/node_modules/electron/dist/Electron.app/Contents/MacOS/Electron"` and this cannot be opened for sure.

Is there a chance to support this new feature once it works in a follow-up PR?

CC: @akosyakov